checker/checker.ml

```diff
@@ -267,26 +267,26 @@ let explain_exn = function
       | Generalization _ -> str"Generalization"
       | ActualType _ -> str"ActualType"
       | CantApplyBadType ((n,a,b),(hd,hdty),args) ->
-          Format.printf "====== ill-typed term ====@\n";
-          Format.printf "@[<hov 2>application head=@ ";
-          Print.print_pure_constr hd;
-          Format.printf "@]@\n@[<hov 2>head type=@ ";
-          Print.print_pure_constr hdty;
-          Format.printf "@]@\narguments:@\n@[<hv>";
-          Array.iteri (fun i (t,ty) ->
-            Format.printf "@[<hov 2>arg %d=@ " (i+1);
-            Print.print_pure_constr t;
-            Format.printf "@ type=@ ";
-            Print.print_pure_constr ty) args;
-          Format.printf "@]@\n====== type error ====@\n";
-          Print.print_pure_constr b;
-          Format.printf "@\nis not convertible with@\n";
-          Print.print_pure_constr a;
-          Format.printf "@\n====== universes ====@\n";
-          chk_pp
-            (Univ.pr_universes
-            (ctx.Environ.env_stratification.Environ.env_universes));
-          str "\nCantApplyBadType at argument " ++ int n
+        (* This mix of printf / pp was here before... *)
+        let fmt = Format.std_formatter in
+        let open Format in
+        let open Print in
+        fprintf fmt "====== ill-typed term ====@\n";
+        fprintf fmt "@[<hov 2>application head=@ %a@]@\n" print_pure_constr hd;
+        fprintf fmt "@[<hov 2>head type=@ %a@]@\n" print_pure_constr hdty;
+        let pp_arg fmt (i,(t,ty)) = fprintf fmt "@[<hov 2>arg %d=@ %a@ type=@ %a" (i+1)
```

Comment:
This is missing a box closing between arguments, right?

Reply:
Well-spotted, thanks.